src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -103,9 +105,12 @@ LogicalMessage[] Extract(IncomingMessage physicalMessage)
                 }
             }
 
+         
             var messageTypes = messageMetadata.Select(metadata => metadata.MessageType).ToList();
             var messageSerializer = deserializerResolver.Resolve(physicalMessage.Headers);
 
+            mapper.EnsureTypesAreInitialized(messageTypes);
```

Comment:
Mapper is used by all the serializers so this is to avoid having them remember to call this? or did you mean to move this to L120?